src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -172,20 +172,13 @@ public void Reanalyze(IIncrementalAnalyzer analyzer, ReanalyzeScope scope, bool
                 }
             }
 
-            private void OnActiveDocumentChanged(object? sender, DocumentId? activeDocumentId)
+            private void OnActiveDocumentSwitched(object? sender, DocumentId? activeDocumentId)
             {
-                var solution = _registration.GetSolutionToAnalyze();
-                if (solution.GetProject(activeDocumentId?.ProjectId) is not { } activeProject)
+                if (activeDocumentId == null)
                     return;
 
-                RoslynDebug.AssertNotNull(activeDocumentId);
-                var analysisScope = SolutionCrawlerOptions.GetBackgroundAnalysisScope(activeProject);
-                if (analysisScope == BackgroundAnalysisScope.ActiveFile)
-                {
-                    // When the active document changes and we are only analyzing the active file, trigger a document
-                    // changed event to reanalyze the newly-active file.
-                    EnqueueFullDocumentEvent(solution, activeDocumentId, InvocationReasons.DocumentChanged, nameof(OnActiveDocumentChanged));
-                }
+                var solution = _registration.GetSolutionToAnalyze();
+                EnqueueFullDocumentEvent(solution, activeDocumentId, InvocationReasons.ActiveDocumentSwitched, eventName: nameof(OnActiveDocumentSwitched));
```

Comment:
```suggestion
                EnqueueFullDocumentEvent(solution, activeDocumentId, InvocationReasons.ActiveDocumentChanged, eventName: nameof(OnActiveDocumentChanged));
```

Reply:
I had originally used that name, but that led to confusion as to whether the contents of the active document changed or just the active document got switched. For example, if it was the former, all incremental analyzers would want to handle this event, but given it is actually the latter, only the diagnostic incremental analyzer cares about it. I am going to retain the "switched" term for now for the invocation reasons that are used by incremental analyzers.